### ISSUE TITLE
Better support for Lua

### ISSUE THREAD

**Issue by slundqui** [PetaVision/OpenPV#12]
Now that everyone has gotten their feet wet with Lua (special thanks to @morae for exploring and using this feature), we should start thinking about better integration of Lua within PetaVision. I believe that @morae is working on creating modules for our LCA/SCANN models so that we can easily import existing LCA modules in Lua. However, the current Lua model requires a middle step, as we need to convert Lua to a parameter file before it can be fed into PV. It would be useful to be able to directly feed in a Lua script into PV as a replacement for our current parameter files. This entails a few things; ex, I believe Lua currently does not support parameter sweeping (and the shiny new batch sweeping). I believe there exists C++ bindings for a Lua interpreter that we should be able to call within PV, so it shouldn't be too difficult to directly run Lua internally as a first step.

**Comment by callistabee** [comment]
I added support for BatchSweeps to `PVModule.lua`. It can be used two ways.

Example 1: Manually

```
...
params['"column":outputPath'] = {
  groupType = "BatchSweep";
  "my/output/path/for/batch/rank0/";
  "my/output/path/for/batch/rank1/";
  "my/output/path/for/batch/rank2/";
  "my/output/path/for/batch/rank3/";
  "my/output/path/for/batch/rank4/";
}
...
```

Example 2: With Generator Function

```
...
pv.batchSweep(params, 'column', 'outputPath', 'my/output/path/for/batch/rank%d', 5);
...
```

These two examples will produce equivalent output, although the second is somewhat limited as it can only make sweeps over string parameters that have the rank index appear in one place.

**Comment by slundqui** [comment]
Adding more information to @morae's post. Commits 9981d4871200fe28bc445c7c0f91870be179e96a and 66d852b0165aa470dd940f74f520904e49b7a92a.

**Comment by peteschultz** [comment]
I just added a branch, lua_api, that starts to address this.  I modified PVModules.lua and HyPerCol::outputParams, so that running the .params.lua file assembles the entire params file into a single Lua string, paramsFileString.  In demos/lua_api, there is a simple program that illustrates using the Lua C API to get this string into C++.

Note: demos/lua_api is not intended to be a permanent addition to the demos directory.  It was just a reasonable-enough holding place to put the program until its functionality is added to PVParams.
